seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-http/src/main/java/org/apache/seatunnel/flink/http/source/util/HttpClientUtils.java

```diff
@@ -0,0 +1,312 @@
+     *
+     * @param url    request address
+     * @param params request parameter map
+     * @return http response result
+     * @throws Exception information
+     */
+    public static HttpClientResult doDelete(String url, Map<String, String> params) throws Exception {
+        if (params == null) {
+            params = new HashMap<>(INITIAL_CAPACITY);
+        }
+
+        params.put("_method", "delete");
+        return doPost(url, params);
+    }
+
+    /**
+     * encapsulate request header
+     *
+     * @param params     request header map
+     * @param httpMethod http request method
+     */
+    public static void packageHeader(Map<String, String> params, HttpRequestBase httpMethod) {
+        // encapsulate request header
+        if (params != null) {
+            Set<Entry<String, String>> entrySet = params.entrySet();
+            for (Entry<String, String> entry : entrySet) {
+                // Set to the request header to the HttpRequestBase object
+                httpMethod.setHeader(entry.getKey(), entry.getValue());
+            }
+        }
+    }
+
+    /**
+     * Encapsulate request parameters
+     *
+     * @param params     request parameter map
+     * @param httpMethod http request method
+     * @throws UnsupportedEncodingException exception information
+     */
+    public static void packageParam(Map<String, String> params, HttpEntityEnclosingRequestBase httpMethod) throws UnsupportedEncodingException {
+        // Encapsulate request parameters
+        if (params != null) {
+            List<NameValuePair> nvps = new ArrayList<>();
+            Set<Entry<String, String>> entrySet = params.entrySet();
+            for (Entry<String, String> entry : entrySet) {
+                nvps.add(new BasicNameValuePair(entry.getKey(), entry.getValue()));
+            }
+
+            // Set to the request's http object
+            httpMethod.setEntity(new UrlEncodedFormEntity(nvps, ENCODING));
+        }
+    }
+
+    /**
+     * get response result
+     *
+     * @param httpClient http client object
+     * @param httpMethod http method onject
+     * @return http response result
+     * @throws Exception information
+     */
+    public static HttpClientResult getHttpClientResult(CloseableHttpClient httpClient, HttpRequestBase httpMethod) throws Exception {
+        // execute request
+        try (CloseableHttpResponse httpResponse = httpClient.execute(httpMethod)) {
+            // get return result
+            if (httpResponse != null && httpResponse.getStatusLine() != null) {
+                String content = "";
+                if (httpResponse.getEntity() != null) {
+                    content = EntityUtils.toString(httpResponse.getEntity(), ENCODING);
+                }
+                return new HttpClientResult(httpResponse.getStatusLine().getStatusCode(), content);
+            }
+        }
+        return new HttpClientResult(HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
This line is just used to make the compiler pass, so you can just add a comment here and return null, then you can remove this constructor.

Reply:
Is it better to return a concrete object than to return null?

Reply:
In this case, we can just return null, since the line 292 will never reach, right? Anyway, this is not a problem, I just want to remove the constructor `HttpClientResult(int code)`.